src/constraints/battery_degradation.jl

```diff
@@ -196,11 +186,12 @@ function add_degradation(m, p; b="ElectricStorage")
         )
         # add augmentation cost to objective
         # maintenance_cost_per_kwh must have length == length(days) - 1, i.e. starts on day 2
+
+        add_to_expression!(m[:TotalStorageCapCosts], p.third_party_factor*(m[:degr_cost]))
+
     else
         @error "Battery maintenance strategy $strategy is not supported. Choose from augmentation and replacement."
     end
-
-    @objective(m, Min, m[:Costs] + m[:degr_cost])
     
     # NOTE adding to Costs expression does not modify the objective function
```

Comment:
@zolanaj does the note on this line mean anything? Is `add_to_expression!` the same as adding degradation costs to `objective` function? When I test with adding degradation cost to objective, I get no replacements for a scenario which gives replacements otherwise.

Reply:
@rathod-b I believe this comment is intended to say that if you change `m[:Costs]` after this line, then it doesn't change the objective and the expression and objective function won't match anymore.

Good to know on the SOC objective; I agree that should be removed if we're running degradation.

Reply:
addressed in https://github.com/NREL/REopt.jl/commit/1b2cfff04bc79fb7a51c06462116e70eeaa0576e